flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/rules/FlinkRuleSets.scala

```diff
@@ -39,18 +39,14 @@ object FlinkRuleSets {
     SubQueryRemoveRule.JOIN)
 
   /**
-    * Handles proper conversion of correlate queries with temporal table functions
-    * into temporal table joins. This can create new table scans in the plan.
+    * Expand plan by replacing references to tables into a proper plan sub trees. Those rules
+    * can create new plan nodes.
     */
-  val TEMPORAL_JOIN_RULES: RuleSet = RuleSets.ofList(
-    LogicalCorrelateToTemporalTableJoinRule.INSTANCE
-  )
+  val EXPAND_PLAN_RULES: RuleSet = RuleSets.ofList(
```

Comment:
It seems to me like providing just a configuration for those rules seems like a halfmassure, since optimizer parameters are just as important as the rules itself (like matching order (`BOTTOM_UP` or `TOP_DOWN`) or vulcano vs hep sequential vs hep simultaneous.

Secondly I'm not sure why someone would want to play/change temporal rules but not changing `TABLE_SUBQUERY_RULES` or `TABLE_REF_RULES`? And I see no reason why someone would like to turn off/disable temporal join rules in the first place - if you are not using temporal table function temporal rule will not fire.

I think that we should either:

1. restrict configuration into just the simplest adding vulcano based rules
2. invest more time into a better configuration method of the rules, that covers all of the possible use cases: matching order, vulcano vs hep, grouping rules together, enforcing that some clean up rule is applied after a group of other rules, etc etc.
3. tell the advance users that want to play with optimiser rules: just override `opitmize()` method and/or create your custom `TableEnvironment`.

I would lean toward the combination of 1. and 3., thus I would drop this issue of configuring `EXPAND_PLAN_RULES`